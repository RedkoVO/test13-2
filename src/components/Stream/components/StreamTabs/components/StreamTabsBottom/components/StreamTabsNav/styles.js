export default () => ({
  root: {
    width: '220px',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  streamTabsNavItem: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    height: '35px',
    textDecoration: 'none',
    padding: '0 12px 0 50px',
    boxSizing: 'border-box',
    fontSize: '13px',
    fontWeight: 700,
    borderRadius: '5px',
    backgroundColor: 'rgba(0,0,0,0.26)',
    color: '#fff',
    marginBottom: '5px',
    cursor: 'pointer'
  },
  liveNow: {
    borderTopLeftRadius: '25px',
    position: 'relative',

    '&::before': {
      position: 'absolute',
      content: `''`,
      backgroundColor: '#ff2e2e',
      borderRadius: '50%',
      top: '50%',
      transform: 'translateY(-50%)',
      left: '20px',
      width: '10px',
      height: '10px'
    }
  },
  streamTabsNavImg: {
    position: 'absolute',
    maxWidth: '20px',
    maxHeight: '20px',
    marginRight: '10px',
    left: '15px'
  },
  streamTabsWatch: {
    position: 'absolute',
    right: '15px'
  }
})
