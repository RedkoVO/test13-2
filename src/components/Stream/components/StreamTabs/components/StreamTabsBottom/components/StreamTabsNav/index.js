import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'
import cn from 'classnames'

import GamepadIcon from '../../../../../../../../assets/images/svg/gamepad.svg'
import Dota2Icon from '../../../../../../../../assets/images/svg/dota-2.svg'
import StarcraftIcon from '../../../../../../../../assets/images/starcraft.png'
import OverwatchIcon from '../../../../../../../../assets/images/overwatch.png'
import CsgoIcon from '../../../../../../../../assets/images/csgo.png'

import styles from './styles'

const StreamTabsNav = ({ classes }) => (
  <div className={classes.root}>
    <div className={cn(classes.streamTabsNavItem, classes.liveNow)}>
      <span className={classes.streamTabsNavText}>LIVE NOW</span>
      <span className={classes.streamTabsWatch}>2</span>
    </div>
    <div className={classes.streamTabsNavItem}>
      <img
        src={GamepadIcon}
        className={classes.streamTabsNavImg}
        alt=""
      />
      <span className={classes.streamTabsNavText}>ALL GAMES</span>
      <span className={classes.streamTabsWatch}>2</span>
    </div>
    <div className={classes.streamTabsNavItem}>
      <img
        src={StarcraftIcon}
        className={classes.streamTabsNavImg}
        alt=""
      />
      <span className={classes.streamTabsNavText}>STARCRAFT 2</span>
      <span className={classes.streamTabsWatch}>2</span>
    </div>
    <div className={classes.streamTabsNavItem}>
      <img
        src={Dota2Icon}
        className={classes.streamTabsNavImg}
        alt=""
      />
      <span className={classes.streamTabsNavText}>DOTA 2</span>
      <span className={classes.streamTabsWatch}>2</span>
    </div>
    <div className={classes.streamTabsNavItem}>
      <img src={OverwatchIcon} className={classes.streamTabsNavImg} alt="" />
      <span className={classes.streamTabsNavText}>CS:GO</span>
      <span className={classes.streamTabsWatch}>2</span>
    </div>
    <div className={cn(classes.streamTabsNavItem, classes.streamTabsNavActive)}>
      <img src={CsgoIcon} className={classes.streamTabsNavImg} alt="" />
      <span className={classes.streamTabsNavText}>CS:GO</span>
      <span className={classes.streamTabsWatch}>2</span>
    </div>
  </div>
)

StreamTabsNav.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(StreamTabsNav)
