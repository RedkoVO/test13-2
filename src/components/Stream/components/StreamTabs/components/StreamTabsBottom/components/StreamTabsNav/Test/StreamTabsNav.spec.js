import * as React from 'react'
import renderer from 'react-test-renderer'

import StreamTabsNav from '../'

it('renders correctly', () => {
  const tree = renderer.create(<StreamTabsNav />).toJSON()
  expect(tree).toMatchSnapshot()
})
