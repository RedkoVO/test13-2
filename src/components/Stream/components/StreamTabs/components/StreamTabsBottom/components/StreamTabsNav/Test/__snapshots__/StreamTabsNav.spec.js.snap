// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StreamTabsNav-root-1"
>
  <div
    className="StreamTabsNav-streamTabsNavItem-2 StreamTabsNav-liveNow-3"
  >
    <span>
      LIVE NOW
    </span>
    <span
      className="StreamTabsNav-streamTabsWatch-5"
    >
      2
    </span>
  </div>
  <div
    className="StreamTabsNav-streamTabsNavItem-2"
  >
    <img
      alt=""
      className="StreamTabsNav-streamTabsNavImg-4"
      src="gamepad.svg"
    />
    <span>
      ALL GAMES
    </span>
    <span
      className="StreamTabsNav-streamTabsWatch-5"
    >
      2
    </span>
  </div>
  <div
    className="StreamTabsNav-streamTabsNavItem-2"
  >
    <img
      alt=""
      className="StreamTabsNav-streamTabsNavImg-4"
      src="starcraft.png"
    />
    <span>
      STARCRAFT 2
    </span>
    <span
      className="StreamTabsNav-streamTabsWatch-5"
    >
      2
    </span>
  </div>
  <div
    className="StreamTabsNav-streamTabsNavItem-2"
  >
    <img
      alt=""
      className="StreamTabsNav-streamTabsNavImg-4"
      src="dota-2.svg"
    />
    <span>
      DOTA 2
    </span>
    <span
      className="StreamTabsNav-streamTabsWatch-5"
    >
      2
    </span>
  </div>
  <div
    className="StreamTabsNav-streamTabsNavItem-2"
  >
    <img
      alt=""
      className="StreamTabsNav-streamTabsNavImg-4"
      src="overwatch.png"
    />
    <span>
      CS:GO
    </span>
    <span
      className="StreamTabsNav-streamTabsWatch-5"
    >
      2
    </span>
  </div>
  <div
    className="StreamTabsNav-streamTabsNavItem-2"
  >
    <img
      alt=""
      className="StreamTabsNav-streamTabsNavImg-4"
      src="csgo.png"
    />
    <span>
      CS:GO
    </span>
    <span
      className="StreamTabsNav-streamTabsWatch-5"
    >
      2
    </span>
  </div>
</div>
`;
